### PR TITLE
Feature: Add option for DynamoDB stream to kick off lambda

### PR DESCRIPTION
**Pull request by bblommers**
Currently, only option for EventSourceMappings is to kick off a Lambda function after receiving an SQS message
This PR adds the option to kick off a Lambda function after updating a DynamoDB table

### CODE REVIEW

**Comment by bblommers** [comment]
Suppose this fixes https://github.com/spulec/moto/issues/815?

File: moto/awslambda/models.py [hunk 1]
```diff
@@ -692,6 +694,18 @@ def create_event_source_mapping(self, spec):
...
+            stream = json.loads(dynamodbstreams_backends[self.region_name].describe_stream(spec['EventSourceArn']))
+            spec.update({'FunctionArn': func.function_arn})
+            esm = EventSourceMapping(spec)
+            self._event_source_mappings[esm.uuid] = esm
+            table_name = stream['StreamDescription']['TableName']
+            table = dynamodb_backends2[self.region_name].get_table(table_name)
+            table.lambda_event_source_mappings[esm.function_arn] = esm
+
+            return esm
+        except Exception:
```
Comment (spulec) [inline moto/awslambda/models.py]: Can we make this more specific to the exceptions that can be thrown here?

File: moto/awslambda/models.py [hunk 2]
```diff
@@ -811,6 +825,19 @@ def send_sns_message(self, function_name, message, subject=None, qualifier=None)
...
         func.invoke(json.dumps(event), {}, {})
 
+    def send_dynamodb_items(self, function_arn, items, source):
+        event = {'Records': [
+            {
+                'eventID': item.to_json()['eventID'],
+                'eventName': 'INSERT',
+                'eventVersion': item.to_json()['eventVersion'],
+                'eventSource': item.to_json()['eventSource'],
+                'awsRegion': 'us-east-1',
```
Comment (spulec) [inline moto/awslambda/models.py]: Should this be `self.region_name`?

File: moto/dynamodb2/models.py [hunk 3]
```diff
@@ -409,6 +409,15 @@ def add(self, old, new):
         seq = len(self.items) + self.starting_sequence_number
         self.items.append(
             StreamRecord(self.table, t, event_name, old, new, seq))
+        result = None
+        from moto.awslambda import lambda_backends
```
Comment (spulec) [inline moto/dynamodb2/models.py]: Any reason to import this inline vs at the module level?

File: tests/test_awslambda/test_lambda.py [hunk 4]
```diff
@@ -1027,6 +1027,54 @@ def test_invoke_function_from_sqs():
...
+        if not log_streams:
+            time.sleep(1)
+            continue
+
+        assert len(log_streams) == 1
+        result = logs_conn.get_log_events(logGroupName='/aws/lambda/testFunction', logStreamName=log_streams[0]['logStreamName'])
+        for event in result.get('events'):
+            if event['message'] == 'get_test_zip_file3 success':
+                return
+        time.sleep(1)
```
Comment (spulec) [inline tests/test_awslambda/test_lambda.py]: What is the sleeping for? Is there a way to make this test without needing them?

File: moto/awslambda/models.py (hunk 2 above)
Comment (bblommers) [reply moto/awslambda/models.py]: Good spot - I've updated it

File: moto/awslambda/models.py (hunk 1 above)
Comment (bblommers) [reply moto/awslambda/models.py]: Rewritten this so that it doesn't throw an exception at all

File: moto/dynamodb2/models.py (hunk 3 above)
Comment (bblommers) [reply moto/dynamodb2/models.py]: I've followed the style in the sqs/models.py, where a lambda function is kicked off whenever an SQS message is received. There the lambda_backends  is also imported locally.
I'm ambivalent where the import happens really - it shouldn't be too resource-heavy.

File: tests/test_awslambda/test_lambda.py (hunk 4 above)
Comment (bblommers) [reply tests/test_awslambda/test_lambda.py]: Again following the SQS example, in the test just above. I guess the Lambda is invoked async when this is run in server mode, so we'd really have to wait for the Lambda to finish executing/writing logs

File: moto/dynamodb2/models.py (hunk 3 above)
Comment (spulec) [reply moto/dynamodb2/models.py]: Ah, no. This is right. Sorry about that.

File: tests/test_awslambda/test_lambda.py (hunk 4 above)
Comment (spulec) [reply tests/test_awslambda/test_lambda.py]: Ah, okay. That is a bit annoying, but I can understand it and don't have a better suggestion off the top of my head.

**Comment by spulec** [comment]
This is great; thanks for helping!

**Pull request merged**